app/controllers/user_reductions_controller.rb

```diff
@@ -22,13 +22,27 @@ def update
     render json: reduction
   end
 
+  def current_user_reductions
+    reducible_type = params[:reducible_type].titleize.singularize
+    reducible_id = params[:reducible_id]
+
+    reductions = UserReduction.where(
+      user_id: credential.current_user_id,
+      reducible_type: reducible_type,
+      reducible_id: reducible_id
+    )
+
+    authorize reductions
```

Comment:
GET requests should do use Pundit's scopes, not `authorize`. Note how `index` doesn't need to call authorize. The Pundit controller integrations will raise if index or show doesn't call a scope.

Reply:
i was really struggling to get this working using pundit scopes but i'll give it a little more time; it's possible the comment you left below will point me in the right direction.

Reply:
Your suggestion worked!